### PR TITLE
[Serve][Doc] Authorization doc

### PR DESCRIPTION
**Pull request by cblmemo**
Adding docs for authorization (introduced in #3552)

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
     - [x] The YAML in the doc
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

**Comment by cblmemo** [comment]
Resolve all comments. This is ready for another look 🫡 @Michaelvll

File: docs/source/serving/auth.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+If the Authorization header is missing or invalid, the service will return a 401 Unauthorized error:
```
Comment (Michaelvll) [inline docs/source/serving/auth.rst:80]: A service client without an API key will not be able to access the service and get a :code:`401 Unauthorized` error:

**Review by Michaelvll — COMMENTED** [review]
Thanks for the update @cblmemo! Left some additional changes.

File: docs/source/serving/auth.rst [hunk 2]
```diff
@@ -74,10 +73,32 @@ With the :code:`--api-key` set, a user's request needs to include an Authorizati
...
+        "finish_reason": "stop",
+        "stop_reason": 128009
+      }
+    ],
+    "usage": {
+      "prompt_tokens": 26,
+      "total_tokens": 160,
+      "completion_tokens": 134
+    }
+  }
```
Comment (Michaelvll) [inline docs/source/serving/auth.rst:99]: Let's wrap this in a drop down:
```suggestion

.. raw:: HTML

  <details>
  
  <summary>Example output</summary>
  
  
.. code-block:: console

  {
    "id": "cmpl-cad2c1a2a6ee44feabed0b28be294d6f",
    "object": "chat.completion",
    "created": 1716819147,
    "model": "meta-llama/Meta-Llama-3-8B-Instruct",
    "choices": [
      {
        "index": 0,
        "message": {
          "role": "assistant",
          "content": "I'm so glad you asked! I'm LLaMA, an AI assistant developed by Meta AI that can understand and respond to human input in a conversational manner. I'm here to help you with any questions, tasks, or topics you'd like to discuss. I can provide information on a wide range of subjects, from science and history to entertainment and culture. I can also assist with language-related tasks such as language translation, text summarization, and even writing and proofreading. My goal is to provide accurate and helpful responses to your inquiries, while also being friendly and engaging. So, what's on your mind? How can I assist you today?"
        },
        "logprobs": null,
        "finish_reason": "stop",
        "stop_reason": 128009
      }
    ],
    "usage": {
      "prompt_tokens": 26,
      "total_tokens": 160,
      "completion_tokens": 134
    }
  }
  
.. raw:: html

  </details>
```

**Review by Michaelvll — APPROVED** [review]
Thanks for the update @cblmemo! LGTM.

**Pull request merged**